test/api_integration/apis/saved_objects_management/scroll_count.ts

```diff
@@ -18,77 +18,132 @@ export default function ({ getService }: FtrProviderContext) {
+          })
+          .expect(200);
+
+        expect(res.body).to.eql({
+          dashboard: 2,
+          search: 1,
+        });
       });
-    });
 
-    it('filters on title when `searchString` is provided', async () => {
-      const res = await supertest
-        .post(apiUrl)
-        .send({
-          typesToInclude: defaultTypes,
-          searchString: 'Amazing',
-        })
-        .expect(200);
-
-      expect(res.body).to.eql({
-        dashboard: 1,
-        visualization: 1,
-        'index-pattern': 0,
-        search: 0,
+      it('filters on title when `searchString` is provided', async () => {
+        const res = await supertest
+          .post(apiUrl)
+          .send({
+            typesToInclude: defaultTypes,
+            searchString: 'Amazing',
+          })
+          .expect(200);
+
+        expect(res.body).to.eql({
+          dashboard: 1,
+          visualization: 1,
+          'index-pattern': 0,
+          search: 0,
+        });
+      });
+
+      it('includes all requested types even when none match the search', async () => {
+        const res = await supertest
+          .post(apiUrl)
+          .send({
+            typesToInclude: ['dashboard', 'search', 'visualization'],
+            searchString: 'nothing-will-match',
+          })
+          .expect(200);
+
+        expect(res.body).to.eql({
+          dashboard: 0,
+          visualization: 0,
+          search: 0,
+        });
       });
     });
 
-    it('includes all requested types even when none match the search', async () => {
-      const res = await supertest
-        .post(apiUrl)
-        .send({
-          typesToInclude: ['dashboard', 'search', 'visualization'],
-          searchString: 'nothing-will-match',
-        })
-        .expect(200);
-
-      expect(res.body).to.eql({
-        dashboard: 0,
-        visualization: 0,
-        search: 0,
+    describe('scroll_count with more than 10k objects', () => {
+      const importVisualizations = async ({
+        startIdx = 1,
+        endIdx,
+      }: {
+        startIdx?: number;
+        endIdx: number;
+      }) => {
+        const fileChunks: string[] = [];
+        for (let i = startIdx; i <= endIdx; i++) {
+          const id = `test-vis-${i}`;
+          fileChunks.push(
+            JSON.stringify({
+              type: 'visualization',
+              id,
+              attributes: {
+                title: `My visualization (${i})`,
+                uiStateJSON: '{}',
+                visState: '{}',
+              },
+              references: [],
+            })
+          );
+        }
+
+        await supertest
+          .post(`/api/saved_objects/_import`)
+          .attach('file', Buffer.from(fileChunks.join('\n'), 'utf8'), 'export.ndjson')
+          .expect(200);
+      };
+
+      before(async () => {
+        await importVisualizations({ startIdx: 1, endIdx: 6000 });
+        await importVisualizations({ startIdx: 6001, endIdx: 12000 });
```

Comment:
As the import limit is `10001`, I'm imported the 12k in two batches. This is also an additional check that the `scroll_count` API works when there are more objects than the limit in the search results.